### PR TITLE
feat(dataZoom): custom brush style in toolbox.feature.dataZoom

### PR DESCRIPTION
**Pull request by zhiyuc123**
## Brief Information

This pull request is in the type of:

- [ ] bug fixing
- [x] new feature
- [ ] others

### What does this PR do?

allow user to custom brush style in toolbox.feature.dataZoom

### Fixed issues

#8964

### Before: What was the problem?

框选型数据区域缩放组件不能更改选框颜色，在暗色主题下表现不好

### After: How is it fixed in this PR?

允许用户通过配置自定义选框样式

       toolbox: {
            feature: {
                dataZoom: {
                    show: true,
                    yAxisIndex: null,
                    brushStyle: {
                        borderWidth: 2,
                        color: 'blue',
                        opacity: 0.1,
                        borderColor: '#596A76',
                     }
                },
            }
        },

### Are there any API changes?

- [ ] The API has been changed.

### Related test cases or examples to use the new APIs

NA.

### Merging options

- [ ] Please squash the commits into a single one when merge.

### CODE REVIEW

File: src/component/toolbox/feature/DataZoom.js [hunk 1]
```diff
@@ -236,16 +240,30 @@ function updateZoomBtnStatus(featureModel, ecModel, view, payload, api) {
...
+    ];
+    var style = {};
+    for (var i = 0; i < properties.length; i++) {
+        var propName = properties[i][1];
+        var val = brushStyle[propName];
+        if (val != null) {
+            style[properties[i][0]] = val;
+        }
+    }
+    return style;
```
Comment (chfw) [inline src/component/toolbox/feature/DataZoom.js:264]: firstly, you may consider es6 for..in syntax to simplify the for loop

secondly, you may consider es6 for..of syntax to shrink your for loop(reduce the number of variables and lines)

lastly, you can try out forEach to completely rewrite the function, i.e.:

```
[
     ['fill', 'color'],
     ['lineWidth', 'borderWidth'],
     ['stroke', 'borderColor'],
     ['opacity', 'opacity']
 ].forEach((item) =>{ console.log(item)})
```

Comment (susiwen8) [reply src/component/toolbox/feature/DataZoom.js:264]: @chfw ECharts do not allow ES6 and ES5 syntax.

Comment (chfw) [reply src/component/toolbox/feature/DataZoom.js:264]: Sorry, I am now aware of it.

File: src/component/toolbox/feature/DataZoom.js [hunk 2]
```diff
@@ -236,16 +240,30 @@ function updateZoomBtnStatus(featureModel, ecModel, view, payload, api) {
             zoomActive
             ? {
                 brushType: 'auto',
-                brushStyle: {
-                    // FIXME user customized?
-                    lineWidth: 0,
-                    fill: 'rgba(0,0,0,0.2)'
-                }
+                brushStyle: mapBrushStyle(featureModel.option.brushStyle)
             }
```
Comment (pissang) [inline src/component/toolbox/feature/DataZoom.js:244]: Model has provided a `getItemStyle` method to map all the style properties.

```js
brushStyle: featureModel.getModel('brushStyle').getItemStyle()
```

**Pull request merged**